doc/user/methods/adaboost_regressor.md

```diff
@@ -0,0 +1,582 @@
+    << "." << std::endl;
+
+// Compute predictions on test data.
+arma::rowvec testPredictions;
+adr.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "Average error on test set: " << testAverageError << "."
+    << std::endl;
+```
+
+---
+
+Load a adaboost regressor and print some information about it.
+
+```c++
+mlpack::AdaBoostRegressor<> adr;
+// This call assumes a adaboost regressor called "adr" has already been saved to
+// `adr.bin` with `data::Save()`.
+mlpack::data::Load("adr.bin", "adr", adr, true);
+
+std::cout << "The adaboost regressor in 'adr.bin' contains " << adr.NumTrees()
+    << " trees." << std::endl;
+if (adr.NumTrees() > 0)
+{
+  std::cout << "The first tree's root node has " << adr.Tree(0).NumChildren()
+      << " children." << std::endl;
+}
+```
+
+---
+
+Train a adaboost regressor on categorical data, and compare its performance 
+with the performance of each individual tree:
+
+```c++
+// Load a categorical dataset.
+arma::mat data;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/telecom_churn.arff.
+mlpack::data::Load("telecom_churn.arff", data, info, true);
+
+arma::rowvec responses;
+// See https://datasets.mlpack.org/telecom_churn.responses.csv.
+mlpack::data::Load("telecom_churn.responses.csv", responses, true);
+
+// Split data into training set (80%) and test set (20%).
+arma::mat trainData, testData;
+arma::rowvec trainResponses, testResponses;
+mlpack::data::Split(data, responses, trainData, testData, trainResponses,
+    testResponses, 0.2);
+
+// Create the tree.
+mlpack::AdaBoostRegressor<> adr;
+// Train on the given dataset, specifying a minimum gain of 1e-7 and keeping the
+// default minimum leaf size.
+const double mse = adr.Train(trainData, info, trainResponses, 20 /*number of trees*/
+    ,10 /* minimum leaf size */, 1e-7 /* minimum gain */, 4 /*maximum depth*/);
+
+// Compute test accuracy for each tree
+arma::rowvec testPredictions;
+for(size_t i = 0; i<adr.NumTrees(); i++)
+{
+  adr.Tree(i).Predict(testData, testPredictions);
+  const double testAverageError = arma::mean(testResponses - testPredictions);
+  std::cout << "Tree " << i << " has averageError: " << testAverageError << "."
+    << std::endl;
+}
+
+// Compute predictions on test data using whole forest.
+adr.Predict(testData, testPredictions);
+
+// Compute the average error on the test set.
+const double testAverageError = arma::mean(testResponses - testPredictions);
+std::cout << "The whole forest has Average error: " << testAverageError << "."
+    << std::endl;
+```
+
+---
+### Advanced Functionality: Template Parameters
+
+#### Fully custom behavior.
+
+The `AdaBoostRegressor` class also supports several template parameters, 
+which can be used for custom behavior during learning. The full signature of 
+the class is as follow:
```

Comment:
I see, so it looks like `AdaBoostRegressor` is specific to decision trees.  But AdaBoost is a general technique that can work with any weak learner (this is why `AdaBoost<>` allows it as a template type).  Shouldn't `AdaBoostRegressor` be able to do the same thing?  e.g. I could have an `AdaBoostRegressor<LinearRegression>` or an `AdaBoostRegressor<LARS>` or something like that.  (...assuming those support weighted learning, which I'm not sure they do.  But still you get my idea, that we should be able to plug in any regression technique that supports weighted learning there.)